inject.js

```diff
@@ -453,6 +453,10 @@ function setupListener() {
       if (coolDown) {
         log("Speed event propagation blocked", 4);
         event.stopImmediatePropagation();
+
+        // Send custom event for other extensions
+        const customEvent = new Event('videoSpeed_ratechange');
```

Comment:
We already dispatch a custom event in: https://github.com/igrigorik/videospeed/blob/113ec8909142099990d43415612fd45f6efc6be3/inject.js#L704-L707

Is that one not sufficient? Anything we can refactor? My intuition is that we should only be dispatching one event.

Reply:
This custom event is being still being accepted by videospeed, which is then stopping propagation.

https://github.com/igrigorik/videospeed/pull/740/files#diff-8dcb72b9c92c5fa9eb8602ce15b90717f8717699b5865db86078df006be5c4c0L451-L455

Reply:
Per the current logic, that should only stop propagation is we're in cool-down phase and intentionally want to block another speed update, and proceed otherwise — right? You should be able to catch this event upstream.. what am I missing?

Reply:
Could the issue be that `refreshCoolDown` is getting called before the `CustomEvent` fires? That could cause `coolDown` to always not be false.

Reply:
After some testing, ratechange events do work when using the YouTube settings, and only don't get passed to my extension when the buttons on the video speed extension are pressed. So, it seems to be an issue with coolDown being true when it should not be.

Could you explain what the purpose of the cool down is?

Reply:
@ChadBailey implemented the logic and can provide full context. The short version though, if I recall correctly, it helps protect from auto-revert that some players invoke after we force a speed update.

I don't see a reason why we can't or shouldn't emit a ratechange event for button presses. That logic should not be subject to cooldown though; we don't want to throttle user-initiated changes.

Reply:
Okay, that's what I was thinking. I was thinking that this custom event would allow videospeed to hide the `ratechange` from the website while still allowing other extensions to know about it. This way, it can still prevent the auto-revert issue,

Reply:
I know this is a wayyy late reply, but I never saw this until now :( I figure I will give the explanation, better late than never.

This effort was done to fix a few problematic sites which would de-apply rate changes directly after applying them since the rate change did not originate from the site itself. If we are finding that this is having unintended side-effects, it may be a good idea to try and implement a second list similar to the disable-list which informs videospeed controller when to enable this behavior.

Reply:
yea, this is what I was thinking. If so, a disable-list like you said could work, or an extra event like the one added here so that it will not be picked up by the site itself, but only by "compatible" extensions